src/promptflow-azure/promptflow/azure/_storage/blob/client.py

```diff
@@ -0,0 +1,109 @@
+import datetime
+import logging
+import threading
+import traceback
+from typing import Optional, Tuple
+
+from azure.ai.ml import MLClient
+from azure.ai.ml._azure_environments import _get_storage_endpoint_from_metadata
+from azure.ai.ml._restclient.v2022_10_01.models import DatastoreType
+from azure.ai.ml.constants._common import STORAGE_ACCOUNT_URLS
+from azure.ai.ml.entities._datastore.datastore import Datastore
+from azure.storage.blob import ContainerClient
+
+from promptflow.exceptions import UserErrorException
+
+datastore_map = {}
+client_map = {}
+_thread_lock = threading.Lock()
+_cache_timeout = 60 * 5  # Will try to refresh default datastore if exceed 5 minutes
+
+
+def get_client(
+    logger: logging.Logger,
+    subscription_id: str,
+    resource_group_name: str,
+    workspace_name: str,
+    credential: Optional[object] = None,
+) -> Tuple[ContainerClient, str]:
+    try:
+        # To write data to blob, user should have "Storage Blob Data Contributor" to the storage account.
+        if credential is None:
+            from azure.identity import DefaultAzureCredential
+
+            credential = DefaultAzureCredential()
+
+        default_datastore = get_default_datastore(subscription_id, resource_group_name, workspace_name, credential)
+
+        storage_endpoint = _get_storage_endpoint_from_metadata()
+        account_url = STORAGE_ACCOUNT_URLS[DatastoreType.AZURE_BLOB].format(
+            default_datastore.account_name, storage_endpoint
+        )
+
+        from azure.ai.ml.constants._common import LONG_URI_FORMAT
```

Comment:
Is there a specific rationale for importing at the function level? I generally prefer imports at the top of the file. :)

Reply:
Fixed. Thanks.